### DISCUSSION TITLE
Is there any benefits to run BERTopic using TPU on Google Colab?

### DISCUSSION THREAD

**Discussion by ArtemBernatskyy** [MaartenGr/BERTopic#1012] [Q&A]
Is there any benefits to run BERTopic using TPU on Google Colab over using GPU instances? Thx!

<img width="396" alt="Captura de pantalla 2023-02-16 a las 20 16 04" src="https://user-images.githubusercontent.com/13070119/219465077-a4fd390c-f2c9-4a43-860f-15597d9cedb2.png">

**Comment by drob-xx** [comment]
~Oh yeah. Significant. Run some tests and you'll see. Also~ cuML is amazing.

**Comment by ArtemBernatskyy** [comment]
Hm, tested this on Google Colab Pro+ on the dataset (500k tweets),
GPU Premium worked 3-5 times faster than TPU Premium,

somehow even loading data (where it writes something about "batches") takes much longer on TPU (GPU: 15min vs TPU: 2hours) and then crunching data is faster on GPU too (GPU: 25min vs TPU: 32min)...

Maybe ... there is some benefit for super large datasets but for 500k tweets dataset GPU Premium works much faster compared to TPU (GPU: 15min inserting data + 25min crunching=40min vs TPU:2hours inserting data + 32min crunching=152min).

Also this could be related to the fact that GPU Premium gives us A100 GPU which overperforms all optimisations which TPU has...
P.S. even colab shows that when using GPU Premium you pay 13credits/hour and when using TPU Premium 2credits/hour...

**Reply by drob-xx** [reply]
Erghhhh... I misread your original - bad advice - my bad. I thought you meant upping to the GPU, I tried the TPU sometime ago and didn't see any improvement - really sorry (hangs head in shame...)
